### ISSUE TITLE
Passing the auth-user as a global param from the controller

### ISSUE THREAD

**Issue by shuroo** [schaloner/deadbolt-2-scala#25]
Hi,

I'm using deadbolt-2-scala with play framework.
I wish to find an elegant way to pass the auth-user in the session in order to pass this data to the logs.

Thanks,
Shiri.

**Comment by schaloner** [comment]
Hi,

you have access to the subject from the request passed into the controller function - the request is an `AuthenticatedRequest`.  For example, 

    def index = deadbolt.SubjectPresentAction().defaultHandler() { implicit request =>
        request.subject.map(sbj => /* log the request */)
        // express the result
    }

Alternatively, you can log the user in the `getSubject` function of your `DeadboltHandler` - this will let you centralize the logging.

Regards,
Steve

**Comment by shuroo** [comment]
Hi,

Well,

I'm familiar with the above tricks.

The problem is, that I'd like to access the request from the models and bl
packahes as well to get the logs.

This cannot be done unless I pass the request as a function param to the
inner functions. I wish to have a better way than that to access this
information to use the logs.

Thanks.

**Comment by schaloner** [comment]
I've been wracking my brains trying to think of a way to resolve this without passing the request around, but it always comes back to a brute force approach of passing the request around.

**Comment by shuroo** [comment]
Yes, I did not yet managed to resolved it other way.

The assistedinject (Guice) may be a possibility, theoretically, not
successfully tested yet.

Thank you anyway.  I think I am closing this for now

(I solved it partially by initializing and passing a logger class with the
info. not the solution I was hoping for, but the easiest. And it looks fine
).

2016-09-05 9:09 GMT+03:00 Steve Chaloner <notifications@github.com>:
